librz/arch/il/analysis_il.c

```diff
@@ -251,20 +253,37 @@ RZ_API RzAnalysisILStepResult rz_analysis_il_vm_step_while(RZ_NONNULL RzAnalysis
 	}
 
 	RzAnalysisILStepResult res = RZ_ANALYSIS_IL_STEP_RESULT_SUCCESS;
+	ut64 addr = 0;
 	while (cond(vm, user)) {
-		ut64 addr = rz_bv_to_ut64(vm->vm->pc);
+		addr = rz_bv_to_ut64(vm->vm->pc);
 		ut8 code[32] = { 0 };
 		analysis->read_at(analysis, addr, code, sizeof(code));
 		RzAnalysisOp op = { 0 };
 		rz_analysis_op_init(&op);
-		int r = rz_analysis_op(analysis, &op, addr, code, sizeof(code), RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_HINT);
+		int r = rz_analysis_op(analysis, &op, addr, code, sizeof(code), RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_HINT | RZ_ANALYSIS_OP_MASK_DISASM);
 		RzILOpEffect *ilop = r < 0 ? NULL : op.il_op;
 
 		if (ilop) {
 			bool succ = rz_il_vm_step(vm->vm, ilop, addr + (op.size > 0 ? op.size : 1));
 			if (!succ) {
 				res = RZ_ANALYSIS_IL_STEP_IL_RUNTIME_ERROR;
 			}
+
+			if (with_events) {
+				RzStrBuf sb;
+				rz_strbuf_init(&sb);
+				rz_il_op_effect_stringify(op.il_op, &sb, true);
+				rz_strbuf_append(&sb, "\n");
+				il_events(vm->vm, &sb);
+
+				char *il_stmt = rz_strbuf_get(&sb);
+				rz_cons_printf("0x%llx [%x%x%x%x] %s\n"
+					       "0x%llx %s\n",
+					addr, code[0], code[1], code[2], code[3],
+					op.mnemonic, addr, il_stmt);
+				rz_strbuf_fini(&sb);
+			}
+
```

Comment:
Invert the condition

Reply:
It seems to me that the “Invert the condition” here is not appropriate.

Reply:
```c
if (!ilop) {
	res = RZ_ANALYSIS_IL_STEP_INVALID_OP;
	rz_analysis_op_fini(&op);
	break;
}
bool succ = rz_il_vm_step(vm->vm, ilop, addr + (op.size > 0 ? op.size : 1));
if (!succ) {
	res = RZ_ANALYSIS_IL_STEP_IL_RUNTIME_ERROR;
	rz_analysis_op_fini(&op);
	break;
}
```

Reply:
`op` could be moved outside the loop body to reduce `rz_analysis_op_fini(&op);` lines count